sphinx/config.py

```diff
@@ -201,6 +194,13 @@ def __init__(self, dirname, filename, overrides, tags):
                     config[k] = copyright_year_re.sub(r'\g<1>%s' % format_date('%Y'),
                                                       config[k])
 
+    @classmethod
+    def read(cls, confdir, filename, overrides=None, tags=None):
```

Comment:
I have a proof of concept patch ready that allows us to read INI-style configuration files. Could we call this something less generic? Maybe `read_py` or `load_py`?

Reply:
Of the two, I prefer `Config.read_from_conf_py()`. `Config.read_py()` also seems fine though. Alternatively, I could add a `mode` parameter that we set to `ini` or `py` later. Whatever you prefer.

    def read(cls, path, overrides=None, tags=None, mode='py'):